apps/frontend/core-dapp/src/features/withdraw/WithdrawStore.ts

```diff
@@ -15,14 +16,36 @@ export class WithdrawStore {
   }
 
   // eslint-disable-next-line require-await
-  async withdraw(): Promise<{
-    success: boolean
-    error?: string | undefined
-  }> {
-    const { preCTTokenStore } = this.root
-    return this.withdrawalAmountBN !== undefined && this.withdrawalAmountBN.gt(0)
-      ? preCTTokenStore.withdraw(this.withdrawalAmountBN)
-      : { success: false }
+  async withdraw(): Promise<void> {
+    if (
+      this.insufficientBalance ||
+      this.withdrawalAmountBN === undefined ||
+      this.withdrawalAmountBN.eq(0)
+    )
+      return
+
+    this.withdrawing = true
+    const { error } = await this.root.preCTTokenStore.withdraw(this.withdrawalAmountBN)
+
+    if (error) {
+      this.root.toastStore.errorToast('Withdrawal failed', error)
+    } else {
+      this.root.toastStore.successToast('Withdrawal was successful 🎉')
```

Comment:
```suggestion
      this.root.toastStore.errorToast('The withdrawal failed', error)
    } else {
      this.root.toastStore.successToast('The withdrawal was successful 🎉')
```

Some minor nit on the copies, it feels more natural to me, but feel free to disregard.